### ISSUE TITLE
Summons dying on their turn not advancing the turn + summon condition expiration timing

### ISSUE THREAD

**Issue by datenshi888** [Lurkars/gloomhavensecretariat#307]
A small thing, but if a summon dies during its turn, say from a retaliate, the app does not advance the turn to the next summon (only applicable if there are multiple summons by the same character at the same time). Doubleclicking/holding the summon to activate its turn works fine.

Additionally, if a summon has a condition on it that is supposed to expire after its turn (strengthen etc.), it doesn't expire until the end of the summoner's turn. That matters in situations where the summoner grants their summon an action during their turn.

**Comment by Lurkars** [comment]
Good point, thanks will check.

**Comment by Lurkars** [comment]
Hopefully solved in  [v0.66.3](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.66.3) (with fixing #309 )

**Comment by datenshi888** [comment]
Summons dying looks to be fixed!

Summon conditions still expire at the end of their summoner's turn instead of their own.

**Comment by Lurkars** [comment]
Ah I think I forgot to add this to the double click toggling. When using tab or character portrait, it's working fine for me. Will fix.

**Comment by Lurkars** [comment]
Hopefully now fully working in  [v0.66.4](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.66.4)

**Comment by datenshi888** [comment]
Hmm. This seems to have introduced or revealed a new bug.

It works fine if there is only 1 summon.

With multiple summons, the first summon works correctly. The second summon still has the condition expiry bug, but otherwise activates and ends its turn correctly. Any summons after the 2nd one do not get activated. Tried it with double clicking and tabbing to do their turns.

**Comment by Lurkars** [comment]
Can only reproduce for last summon. You may share your setup? So character with which summons and which conditions.

Now fix for last summon in [v0.66.5](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.66.5). Other issue cannot reproduce, but maybe I am assuming a wrong setup.

**Comment by datenshi888** [comment]
I've been testing with Coral and Snowflake and a random monster or Scenario 0. Summons were 2 White Owls, Snow Fox and Polar Cat in that order. I've pretty much only tested with Strengthen.

I've shuffled the summons around a bit and it looks like it might be an issue specifically with the White Owl summons. After summoning a Snow Fox first and the Owls after, Fox and both Owls worked fine, but the Cat didn't activate its turn.

In the middle of writing this I restarted Scenario 0 to revert back to the original setup of 2 Owls, Fox and Cat and was suddenly no longer able to reproduce the issue. It is possible that I refreshed or hard refreshed the page in between but I don't remember doing it. I did make sure the page was updated to 0.66.5 before I started testing.
